### PR TITLE
feat(tiller): Add .Release.Revision and .IsUpgrade

### LINKED ISSUE #1554: Need ability in templates to detect whether the current release is an upgrade
A good amount of charts are randomly generating passwords when they are run. This causes issues when upgrading charts, causing passwords to be regenerated. 

What I would like to see is something along the lines of:

{{- if .Release.Upgrade }}
{{- end }}

Another approach could be to expose the release number:
{{- if .Release.Revision == 1 }}
{{- end }}

### PR DESCRIPTION
**Pull request by technosophos**
New properties are available to template developers.

Closes #1554

### CODE REVIEW

File: docs/chart_template_guide/builtin_objects.md [hunk 1]
```diff
@@ -11,6 +11,8 @@ In the previous section, we use `{{.Release.Name}}` to insert the name of a rele
 	- `Release.Time`: The time of the release
 	- `Release.Namespace`: The namespace to be released into (if the manifest doesn't override)
 	- `Release.Service`: The name of the releasing service (always `Tiller`).
+  - `Release.Revision`: The revision number of this release. It begins at 1 and is incremented for each `helm upgrade`.
```
Comment (chancez) [inline docs/chart_template_guide/builtin_objects.md]: Tabs vs spaces. Different from the ones above in the markdown.

**Comment by chancez** [comment]
Perhaps it would be useful to also add `.IsInstall`. It seems like that would be the more common use case and would be more ergonomic than `not .IsUpgrade`.

File: docs/chart_template_guide/builtin_objects.md (hunk 1 above)
Comment (technosophos) [inline docs/chart_template_guide/builtin_objects.md]: Thanks.

**Comment by technosophos** [comment]
I'm adding IsInstall too at @chancez's suggestion.

**Comment by adamreese** [comment]
manually tested

**Pull request merged**